pkg/executables/cmk.go

```diff
@@ -0,0 +1,308 @@
+package executables
+
+import (
+	"bytes"
+	"context"
+	_ "embed"
+	"encoding/json"
+	"fmt"
+	"path/filepath"
+	"strconv"
+	"time"
+
+	"github.com/aws/eks-anywhere/pkg/filewriter"
+	"github.com/aws/eks-anywhere/pkg/logger"
+	"github.com/aws/eks-anywhere/pkg/retrier"
+	"github.com/aws/eks-anywhere/pkg/templater"
+)
+
+//go:embed config/cmk.ini
+var cmkConfigTemplate string
+
+const (
+	cmkConfigFileName = "cmk_tmp.ini"
+	maxRetriesCmk     = 5
+	backOffPeriodCmk  = 5 * time.Second
+)
+
+type Cmk struct {
+	writer     filewriter.FileWriter
+	executable Executable
+	retrier    *retrier.Retrier
+}
+
+// TODO: Add support for domain, account filtering
+func (c *Cmk) ValidateTemplatePresent(ctx context.Context, config CmkExecConfig, domain, zone, account, template string) error {
```

Comment:
If you want to simplify these interfaces and you don't mind tiding an instance of `Cmk` to a particular set of CloudStack Cluster and creds (which I think it's ok), you could just pass this to the constructor and have store in `Cmk`